hack/log/log-dump.sh

```diff
@@ -14,7 +14,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -o errexit
```

Comment:
why does this need to be removed? Instead of doing this, it might be better to change the commands that we want to run as best effort with `|| echo "failed to do xyz"`.

Reply:
Added https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/683/files#diff-a9012e009906a8e18d16f7b23dfc67e1R129